bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,277 @@
+        Updates the class with new attributes func_call and func_sign to hold them,
+        """
+        func = self.func
+        if func.__name__ == '<lambda>' : 
+            func.__name__ = f"_lambda_func_number_{FunctionJitter.lambda_function_counter}"
+            FunctionJitter.lambda_function_counter += 1
+        self.func_call = f"{func.__name__}({', '.join(str(arg_info[1]) for arg_info in self.args_info.values())})"
+        self.func_sign = [str(self.args_info[p][0]) for p in self.params]
+
+            
+    def jit_function(self, func, cols_list, extra_args):
+        """
+        Jits the provided function using ROOT's NumbaDeclare.
+        Also checks if the function was jitted earlier in which case it won't jit again but if signature does not match. It raises an error.
+
+        Arguments:
+        func: A python callable
+        cols_list: A list of columns of RDF on which func depends on.
+        extra_args: A dict of extra arguments that func requires.
+        """
+        if func.__name__ in FunctionJitter.function_cache:
+            func_call, func_sign = FunctionJitter.function_cache[func.__name__]
+            self.find_function_params(func)
+            self.generate_func_args(cols_list, extra_args)
+            self.find_function_signature()
+            self.generate_function_call()
+            if self.func_sign != func_sign:
+                raise ValueError("Trying to re-use a function. Do not change function signature.".format(func))
+            return self.func_call
+        self.find_function_params(func)
+        self.generate_func_args(cols_list, extra_args)
+        self.find_function_signature()
+        self.generate_function_call()
+        FunctionJitter.function_cache[self.func.__name__] = (self.func_call, self.func_sign)
+        _NumbaDeclareDecorator(self.func_sign, self.return_type)(self.func)
+        return self.func_call
+
+def _PyFilter(rdf, callable_or_str, *args , extra_args = {} ):
+    """
+    Filters the entries of RDF according to a given condition.
+    Arguments:
+        callable_or_str: The condition according to which the RDF is to be filtered. 
+            It can be either a python callable or a c style string.
+        *args:
+            Can be at most two of them.
+            a. The list of columns in  which case the starting arguments of the provided function callable in mapped to the columns in the list.
+                If not provide then it tries maps the name of the parameter to a column name of the RDF. Flags an error if not found.
+            b. The name of the Filter as a string.
+        extra_args: If the function needs any other parameters that are not columns and must be passed explicitly.
+    Returns:
+        self: The filtered instance of the RDF.
+    
+    Examples:
+    1. rdf.Filter("x>y")
+        Filter using a string containing c++ code 
```

Comment:
I would just include examples of what this pythonization brings on the table. This was already possible, so I wouldn't include it.